scIB/metrics.py

```diff
@@ -710,14 +710,24 @@ def pc_regression(data, variable, pca_var=None, n_comps=50, svd_solver='arpack',
     else:
         X_pca = matrix
         n_comps = matrix.shape[1]
-    
+
     ## PC Regression
     if verbose:
         print("fit regression on PCs")
-    
-    # one-hot encode categorical values
-    variable = pd.get_dummies(variable).to_numpy()
-    
+
+    # handle categorical values
+    try:
+        categorical = variable.dtype.name == 'category'
+    except:
+        categorical = not isinstance(variable[0], (int, float))
+
+    if categorical:
+        if verbose:
+            print("one-hot encode categorical values")
+        variable = pd.get_dummies(variable).to_numpy()
+    else:
+        variable = np.array(variable).reshape(-1, 1)
```

Comment:
why do you reshape here?

Reply:
To make it compatible with lm. I would otherwise get an error. Any alternative suggestions?

Reply:
All good... i thought a transpose or sth should work as well, but that doesn't work for me ^^.